bundle/src/main/java/com/adobe/acs/commons/redirects/servlets/ImportRedirectMapServlet.java

```diff
@@ -126,9 +133,23 @@ void update(Resource root, Collection<RedirectRule> rules) throws PersistenceExc
             }
             props.put(RedirectRule.CONTEXT_PREFIX_IGNORED, rule.getContextPrefixIgnored());
             props.put(PROPERTY_RESOURCE_TYPE, REDIRECT_RULE_RESOURCE_TYPE);
-            props.put(JCR_CREATED, Calendar.getInstance());
-            props.put(JCR_CREATED_BY, resolver.getUserID());
-            resolver.create(root, "redirect-rule-" + (++idx), props);
+
+            Resource redirect = jcrRedirects.get(rule.getSource());
+            if(redirect == null){
+                // add mix:created, AEM will initialize jcr:created and jcr:createdBy from the current session
+                props.put(JCR_MIXINTYPES, MIX_CREATED);
+                String nodeName = ResourceUtil.createUniqueChildName(root, "redirect-rule-");
+                resolver.create(root, nodeName, props);
+            } else {
+                // add mix:lastModified so that AEM updates jcr:lastModified and jcr:lastModifiedBy
+                ValueMap valueMap = redirect.adaptTo(ModifiableValueMap.class);
```

Comment:
if you don't have write permissions in this node, this will return ``` null``` and the next line will throw a NPE. You should add a check here and abort the whole process if the user does not have this permission.

Reply:
the code now throws PersistenceException, same exception as would happen when creating redirects via resourceResolver.create